### PR TITLE
Build: add database items to homepage

### LINKED ISSUE #4: Items from database
image

- Each carousel should have more than one item inside, and go to Product search page/ Buyer's Product Messages Page ON PROGRESS when clicked on

### PR DESCRIPTION
**Pull request by udenizdemirbilek**
closes #4 

Add items from database to homepage as cards, there are links underneath the cards to forward to
product search page/ Buyer's Product Messages Page ON PROGRESS.
Screenshot test for HomePageItems provided, App test has been changed to reflect new content.

### CODE REVIEW

**Review by ceyhungulbas — CHANGES_REQUESTED** [review]
Ellerine saglik Deniz! Looks like you really worked on this. Really good job. With your comments I think functionality will work with firebase. 

I just want to suggest only one thing: A little CSS improvement -card's background color, border radius, white lines at the edge of the text- can be good.

*The pic I showed is about card's background color.

resim

File: public/index.html [hunk 1]
```diff
@@ -24,7 +24,8 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+  
+    <title>Giftify</title>
```
Comment (halitbatur) [inline public/index.html]: this should be in a seperate PR

File: public/manifest.json [hunk 2]
```diff
@@ -1,23 +1,7 @@
 {
-  "short_name": "React App",
-  "name": "Create React App Sample",
-  "icons": [
-    {
-      "src": "favicon.ico",
-      "sizes": "64x64 32x32 24x24 16x16",
-      "type": "image/x-icon"
-    },
-    {
```
Comment (halitbatur) [inline public/manifest.json]: same with this file this should have been done in another PR

**Review by halitbatur — CHANGES_REQUESTED** [review]
I didn't take much look cause indentation needs to be fixed, please make sure to edit the ```yarn run fix``` script in order to check for ```scss``` extenion files instead of ```css```

**Comment by udenizdemirbilek** [comment]
Fixed issues Please Take Another Look @halitbatur @ceyhungulbas

**Review by ceyhungulbas — APPROVED** [review]
Deniz first of all im so sorry cause I didnt see the new version. Sorry for delay... Ellerine saglik. I just checked and looks good to me!

File: src/components/HomePageItemCard/HomePageItemCard.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { Card } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import coffeepot from "../../images/coffeepot.jpg";
+import "./HomePageItemCard.css";
+
+function HomePageItemCard({ itemName = "", id }) {
+  return (
+    <Card className="homePageItemCard">
+      <Card.Img className="itemImage mx-auto" variant="top" src={coffeepot} />
```
Comment (halitbatur) [inline src/components/HomePageItemCard/HomePageItemCard.jsx]: I think here in the src it should be coming from the props too in case you have different items with different images, and you can also have a deafult image in case no image is passed

File: src/components/HomePageItems/HomePageItems.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { Row, Col } from "react-bootstrap";
+import "./HomePageItems.css";
+import HomePageItemCard from "../HomePageItemCard/HomePageItemCard";
+
+function HomePageItems() {
+  const exampleItems = [
```
Comment (halitbatur) [inline src/components/HomePageItems/HomePageItems.jsx]: This array should be coming from HomePage component via props and then used to render the items

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/HomePageItemCard/HomePageItemCard.jsx (hunk 3 above)
Comment (udenizdemirbilek) [reply src/components/HomePageItemCard/HomePageItemCard.jsx]: You're totally right!

**Comment by udenizdemirbilek** [comment]
Fixed issues Please Take Another Look @halitbatur @ceyhungulbas
Also added read functions from firebase, item id is passed down as prop to RequestItemPage

File: src/App.js [hunk 5]
```diff
@@ -1,25 +1,24 @@
...
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <Container fluid>
+        <Switch>
+          <Route exact path="/" component={HomePage} />
+          <Route path="/request-item/:id">
```
Comment (halitbatur) [inline src/App.js]: lets use exact path here in order to make sure we dont create a page called `/request-item/5/<anything>`

File: src/components/HomePageItems/HomePageItems.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Row, Col } from "react-bootstrap";
+import "./HomePageItems.css";
+import HomePageItemCard from "../HomePageItemCard/HomePageItemCard";
+
+function HomePageItems({ dbItems }) {
+  return (
+    <Row xs={2} md={3} xl={4} className="homePageItems">
+      {dbItems.map((e) => (
```
Comment (halitbatur) [inline src/components/HomePageItems/HomePageItems.jsx]: please lets use a better variable name instead of `e`

File: src/containers/HomePage/HomePage.jsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from "react";
+import { Container } from "react-bootstrap";
+import HomePageItems from "../../components/HomePageItems/HomePageItems";
+import "./HomePage.css";
+import db from "../../firebase";
+
+// After deciding on the database structure,the example array should be deleted and
+//  replaced with a function that reads from firebase, takes 8 items by random
```
Comment (halitbatur) [inline src/containers/HomePage/HomePage.jsx]: extra space here

File: src/containers/HomePage/HomePage.jsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+// and passes it to HomePageItems as a prop
+
+function HomePage() {
+  const [dbItems, setDbItems] = useState([]);
+
+  useEffect(() => {
+    db.collection("homePageItems")
```
Comment (halitbatur) [inline src/containers/HomePage/HomePage.jsx]: lets use async await instead of .then()

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/HomePageItems/HomePageItems.jsx (hunk 6 above)
Comment (udenizdemirbilek) [reply src/components/HomePageItems/HomePageItems.jsx]: used dbItem

File: src/containers/HomePage/HomePage.jsx (hunk 7 above)
Comment (udenizdemirbilek) [reply src/containers/HomePage/HomePage.jsx]: wow you have the eyes of a hawk! :)

**Comment by udenizdemirbilek** [comment]
fixed latest issues, 
fetching from firebase uses async await now
PTAL

**Comment by ceyhungulbas** [comment]
Deniz I just check it and LGTM. Functionality is working, styling is good, folder structure is delicious! 👌🏼 Ellerine saglik abi.

**Review by ceyhungulbas — APPROVED** [review]
Deniz I just check it and LGTM. Functionality is working, styling is good, folder structure is delicious! 👌🏼 Ellerine saglik abi.

**Review by busesisi — COMMENTED** [review]
The css and functionality looks very good Deniz. I see some tests are failing but tother than that, ellerine sağlık :)

**Comment by udenizdemirbilek** [comment]
fixed latest issues,
fetching from firebase uses async await now
PTAL @halitbatur

**Comment by halitbatur** [comment]
@udenizdemirbilek 
image
I guess ur new set up is dealing with end of line differently than wsl and that causes this bug where u have a lot of lines extra in the changed lines, please take a look here : https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings to fix the issue

**Comment by udenizdemirbilek** [comment]
@halitbatur PTAL line endings are fixed, fingers are crossed

**Review by halitbatur — APPROVED** [review]
good work !

**Comment by udenizdemirbilek** [comment]
> good work !

This is a small step for mankind but a big step for me :D
